remote-config/src/main/kotlin/com/rakuten/tech/mobile/remoteconfig/ConfigCache.kt

```diff
@@ -38,28 +38,19 @@ internal class ConfigCache @VisibleForTesting constructor(
     private var configBody = applyConfig()
 
     init {
-        poller.start {
-            try {
-                val fetchedConfig = fetcher.fetch()
-
-                verifier.ensureFetchedKey(fetchedConfig.keyId)
-
-                if (verifier.verify(fetchedConfig)) {
-                    file.writeText(fetchedConfig.toJsonString())
-                }
-                if (applyDirectly) {
-                    configBody = applyConfig()
-                }
-            } catch (error: Exception) {
-                Log.e("RemoteConfig", "Error while fetching config from server", error)
-            }
-        }
+        startPoller()
     }
 
     operator fun get(key: String) = configBody[key]
 
     fun getConfig(): Map<String, String> = configBody
 
+    fun fetchConfig(listener: FetchConfigCompletionListener) {
+        // For resetting the polling delay.
+        poller.stop()
+        startPoller(listener)
```

Comment:
Maybe we don't need to reset the poller if it's causing difficulties and issues? I think it's more ideal that the poller is reset, but it's not really a big deal if we can't reset it.

Maybe you could add a `reset` function to `AsyncPoller`? I think that would mean you would need to save a reference to the method that's passed to `start`.

Reply:
changed to reset and used the previously executed method